Firestore/core/src/firebase/firestore/core/relation_filter.cc

```diff
@@ -54,15 +54,15 @@ bool RelationFilter::MatchesValue(const FieldValue& other) const {
 bool RelationFilter::MatchesComparison(const FieldValue& other) const {
   switch (op_) {
     case Operator::LessThan:
-      return value_ < other;
+      return other < value_;
```

Comment:
Optional: this looks a little counterintuitive now that the order has been changed, due to the names. Unfortunately, I don't really have a good suggestion, but even renaming `value_` to `rhs_`, though perhaps not elegant, would at least explain the ordering. (another rough idea is `compare_what < compare_to_`).